### ISSUE TITLE
Cannot run Llama.cpp on Apple Silicon M3 Max

### ISSUE THREAD

**Issue by dineshdharme** [ggerganov/llama.cpp#4440]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

# Expected Behavior

Please provide a detailed written description of what you were trying to do, and what you expected `llama.cpp` to do.

# Current Behavior

Please provide a detailed written description of what `llama.cpp` did, instead.

dyld[6390]: Symbol not found: _cblas_sgemm$NEWLAPACK$ILP64
  Referenced from: <2D4EAB89-571E-3C5D-9B70-4E5AF34E5070> /Users/path/to/llama.cpp/main
  Expected in:     <02B4577F-7DB2-3A40-8800-656D8D30A763> /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
fish: Job 1, './main -m ./models/mistral-7b-i…' terminated by signal SIGABRT (Abort)

ls -al /System/Library/Frameworks/Accelerate.framework
lrwxr-xr-x    1 root  wheel    27 Nov 18 13:02 Accelerate@ -> Versions/Current/Accelerate
lrwxr-xr-x    1 root  wheel    27 Nov 18 13:02 Frameworks@ -> Versions/Current/Frameworks
lrwxr-xr-x    1 root  wheel    26 Nov 18 13:02 Resources@ -> Versions/Current/Resources
drwxr-xr-x    4 root  wheel   128 Nov 18 13:02 Versions/

ls -al /System/Library/Frameworks/Accelerate.framework/Versions/A/
drwxr-xr-x  5 root  wheel  160 Nov 18 13:02 ./
drwxr-xr-x  4 root  wheel  128 Nov 18 13:02 ../
drwxr-xr-x  4 root  wheel  128 Nov 18 13:02 Frameworks/
drwxr-xr-x  7 root  wheel  224 Nov 18 13:02 Resources/
drwxr-xr-x  3 root  wheel   96 Nov 18 13:02 _CodeSignature/

So it seems it is trying to find Accelerate at `/System/Library/Frameworks/Accelerate.framework/Versions/A/` but actually it is located at `/System/Library/Frameworks/Accelerate.framework`

# Environment and Context

Please provide detailed information about your computer setup. This is important in case the issue is not reproducible except for under certain specific conditions.

* Physical (or virtual) hardware you are using, e.g. for Linux:

`$ lscpu`

* Operating System, e.g. for Linux:

`$ uname -a`
MacOS Sonoma

```
$ python3 --version
Python 3.10.6
$ make --version

GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
$ g++ --version
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin23.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

```

# Steps to Reproduce

Basically compile steps for MacOS.

# Failure Logs

**Comment by dineshdharme** [comment]
This bug is being investigated here : 

https://github.com/ml-explore/mlx/issues/264

It seems it is also related to this bug as mentioned in the above issue.

https://github.com/numpy/numpy/issues/25026

**Comment by UnknownLi** [comment]
@dineshdharme  Hello brother, how did you solve this problem in the end? I almost encountered the same problem and system environment.

dyld[83954]: Symbol not found: _cblas_sgemm$NEWLAPACK$ILP64
  Referenced from: <DEEC0347-1CE8-31CF-9AFB-7A250D78F4FB> /Users/x/work/src/github.com/OpenBMB/ollama/llm/llama.cpp/llama-minicpmv-cli
  Expected in:     <375C2A3C-1105-3576-92FF-7710C719D5A7> /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
